frontend/src/components/FindMatch.js

```diff
@@ -134,6 +110,36 @@ const FindMatch = () => {
     )
   }
 
+  matchingSocket.on('found-connection', (username, difficulty) => {
+    const room = {
+      room_id: username,
+      id1: username,
+      id2: user.username,
+      id1_present: true,
+      id2_present: true,
+      difficulty: difficulty.toLowerCase(),
+      datetime: new Date(),
+    }
+    setRoom(room)
+  })
+
+  useEffect(() => {
+    if (room) {
+      try {
+        createRoomService(room)
```

Comment:
This is an async function so we should use `await` on it, otherwise, we open ourselves up for race conditions to happen

Reply:
lemme update it!